### ISSUE TITLE
ASK

### ISSUE THREAD

**Issue by henryprobosantoso** [w86763777/8051-ESP8266-Project#1]
good evening Mr Yi Lun wu, do you have a website with full of explanation for this project. Im so interesting with this project, but i have no idea how to get it. Thank you in advance.

**Comment by w86763777** [comment]
Sorry, there is no other documents about this project.

The project is a guard system. 8051 receive distance measurement from ultrasonic sensor, and send it to TCP server. If the distance smaller than a threshold, the server sends notification to Android app.

Please feel free to let me know if you have any question.

**Comment by henryprobosantoso** [comment]
im sorry mr yi lun wu, what is the type of keyoboard you are use ?

**Comment by w86763777** [comment]
Much like this [one](https://img.ruten.com.tw/s1/9/e6/89/21408097752713_824.jpg), a 4 * 4 membrane keyboard

**Comment by henryprobosantoso** [comment]
Mr yi lun wu, how to use the server with gradle, i dont have any idea about it, can u give me some reference or guidance

**Comment by henryprobosantoso** [comment]
![image](https://user-images.githubusercontent.com/30350164/48478837-09800e80-e838-11e8-86eb-caf558042887.png)

my gradle run stuck, and i dont know what to do, ehmm is that stuck, or is it really long ?

**Comment by w86763777** [comment]
Maybe there is another program using port 45678.
In my experience, is it possible that you forget to shut down the previous program and run a new one.
New program will try to access same port and show this error message because the port is occupied by old one.

**Comment by henryprobosantoso** [comment]
Mr yi lun wu, what kind of mosfet you are used between the RX port from 8051 into CH_PD of ESP8266

**Comment by w86763777** [comment]
This N-Channel MOSFET(2N7000) act as a 3.3v <--> 5v level shifter. You can use any component functioning as TTL to CMOS logic converter to replace it

**Comment by henryprobosantoso** [comment]
Thankyou mr yi lun wu.

Pada tanggal Jum, 23 Nov 2018 10.33 Yi-Lun Wu <notifications@github.com
menulis:

**Comment by henryprobosantoso** [comment]
Mr Yi Lun Wu, how to use the TCP client ?

**Comment by w86763777** [comment]
Sorry for late reply. Do you mean how to "use" or how to "build"?
TCP client is an android app, you need to start the TCP server first and enter server ip/port to app for connection.
Notice, if your server do not have fixed ip, you should make sure that app and server in same LAN.

**Comment by henryprobosantoso** [comment]
thank you mr, i have built the client android app with android studio and succesfull. Now, how to start the TCP server ??

i have tried with gradle run on command promt, and this is the output, am i doing right ? but then i am confused, i dont know what to do then.

![image](https://user-images.githubusercontent.com/30350164/49206678-5ee61f00-f3e5-11e8-93da-46ddf2b4ab3a.png)

**Comment by henryprobosantoso** [comment]
Mr yi lun wu, what is the value of the oscillator you use ? i am using 16mhz oscillator, is there something on the code i have to change ?
thank you

**Comment by w86763777** [comment]
Oscillators is 12MHz, and my 8051 need 12 machine cycle for 1 operation.

1. You need to set the [baud rate](https://github.com/w86763777/8051-ESP8266-Project/blob/master/project/bt_lcm.asm#L612) to 9600.
2. Because I hard code [ip and address](https://github.com/w86763777/8051-ESP8266-Project/blob/master/project/bt_lcm.asm#L1264) in 8051 asm file, you also need to correctly edit it to meet your environment. The default port of server is 45678.
3. For measuring the distance, you need to make sure that [the cycle of timer0](https://github.com/w86763777/8051-ESP8266-Project/blob/master/project/bt_lcm.asm#L1136) be 10us.

How to use:
1. open server
2. start 8051 -> connect to wifi -> (waiting for auto connect to server)
3. open app and "Link Device", identity is defined [here](https://github.com/w86763777/8051-ESP8266-Project/blob/master/project/bt_lcm.asm#L77), you don't need to change it

**Comment by henryprobosantoso** [comment]
sorry mr yi lun wu, i have try everything u said above. But i have checked the tx rx between my 8052 and eesp8266 there is no communication between them. i am suspicious with my calculation of my baud rate setting. i use 2 capacitor ceramic 22pf in pararel and 12MHZ crystal osci. Do i have to change something ?

thank you

**Comment by w86763777** [comment]
Does TX of 8052 have any output? You can use TTL to USB to read TX of 8052 by your computer and treat received message as ascii code, you should see some AT commands.

Actually, i don't know the details of difference between 8052 and 8051.

**Comment by henryprobosantoso** [comment]
Mr yi lun wu, what does this line means ?

//MOV IE, #10010000B        ;Enable serial interrupt

should i use that line ?

**Comment by w86763777** [comment]
No, the system will periodically read data from serial port, so serial interrupt should be disabled.
Actually, all the interrupt is disabled at the beginning. Only when try to measure the distance, the timer 0 interrupt and external 0 interrupt will be enabled. After timeout or it finish measuring the distance, all the interrupt are disabled again.

**Comment by w86763777** [comment]
Can you provide the data sheet of your 8052 ??
If i have time, i will try to figure out how to set register in your chip.

**Comment by henryprobosantoso** [comment]
this is the datasheet of ic im using [https://www.keil.com/dd/docs/datashts/atmel/at89s52_ds.pdf](url) 
In my case i didnt succesfull to connect my AT89S52 to the ESP8266. Im using 12MHZ and 22pF.
thank you mr yi lun wu for your kindly heart.

**Comment by henryprobosantoso** [comment]
hei sir, is there any spesific configuration to my ESP8266 so it can be connected into my at89s52 ?
